web/packages/plugins/shared/common/src/components/TreeView.tsx

```diff
@@ -130,17 +131,16 @@ export const TreeView = (props: {
   }, [])
 
   const _onClick = (node: TreeNode) => {
-    if (!node.expanded) {
+    if (![BlueprintEnum.PACKAGE, 'dataSource'].includes(node.type)) {
+      onSelect(node)
+    } else if (!node.expanded) {
```

Comment:
This seem to have broken collapsable DataSources in the treeView

Reply:
What was the problem this was trying to fix?

Reply:
I got some memory leak issue, but I did not manage to reproduce it, so remove the changes again.